src/lib/utils.ts

```diff
@@ -79,12 +78,17 @@ export function formatNumber(
   return numberFormat.format(standardUnitAmount);
 }
 
-export function TokenAmountDisplay(
-  amount: string | BigNumberish,
-  decimals: number,
-  currency: string,
+export function TokenAmountDisplay({
```

Comment:
Can we combine these methods? Maybe a static inside `TokenAmountDisplay`?

Reply:
Do you mean something like this?
`import { useMemo } from "react";
import { formatNumber } from "@/lib/utils";
import { BigNumberish } from "ethers";
import Tenant from "@/lib/tenant/tenant";
const { token } = Tenant.current();

type TokenAmount = {
  amount: BigNumberish;
  decimals?: number;
  currency?: string;
  maximumSignificantDigits?: number;
};

export function formatTokenAmount({
  amount,
  decimals = token.decimals,
  currency = token.symbol,
  maximumSignificantDigits = 2,
}: TokenAmount) {
  const formattedNumber = formatNumber(
    amount,
    decimals,
    maximumSignificantDigits
  );
  return `${formattedNumber} ${currency}`;
}

export function TokenAmountDisplay({
  amount,
  decimals = token.decimals,
  currency = token.symbol,
  maximumSignificantDigits = 2,
}: TokenAmount) {
  const formattedAmount = useMemo(
    () =>
      formatTokenAmount({
        amount,
        decimals,
        currency,
        maximumSignificantDigits,
      }),
    [amount, decimals, currency, maximumSignificantDigits]
  );

  return <span>{formattedAmount}</span>;
}
`

Reply:
I am going to merge, since I believe this would not make a big difference